sources/httpUtils.ts

```diff
@@ -1,20 +1,49 @@
-import assert          from 'assert';
-import {UsageError}    from 'clipanion';
-import {once}          from 'events';
-import {stderr, stdin} from 'process';
-import {Readable}      from 'stream';
+import assert                     from 'assert';
+import {UsageError}               from 'clipanion';
+import {once}                     from 'events';
+import {stderr, stdin}            from 'process';
+import {Readable}                 from 'stream';
+
+import {DEFAULT_NPM_REGISTRY_URL} from './npmRegistryUtils';
 
 async function fetch(input: string | URL, init?: RequestInit) {
   if (process.env.COREPACK_ENABLE_NETWORK === `0`)
     throw new UsageError(`Network access disabled by the environment; can't reach ${input}`);
 
   const agent = await getProxyAgent(input);
 
+  if (typeof input === `string`)
+    input = new URL(input);
+
+  let headers = init?.headers;
+  const {username, password} = input;
+  if (username || password) {
+    headers =  {
+      ...headers,
+      authorization: `Bearer ${Buffer.from(`${username}:${password}`).toString(`base64`)}`,
```

Comment:
Without testing it but I think that should be `Basic ${Buffer.from(`${username}:${password}`).toString(`base64`)}` instead when using username and password. 🤔 

@aduh95

Reply:
Good catch, thankfully it's already fixed on `main`: https://github.com/nodejs/corepack/pull/454

Reply:
Oh sorry. Okay the fix seems not the be released yet, that might explain why it's not working.